src/asm/stages/stage_1.asm

```diff
@@ -13,18 +14,34 @@ _start:
 	# Setup the stack to be at _stack_end and make
 	# direction of memory processing lowest-to-highest
 	cld
-	mov sp, _stack_end
+	lea sp, _stack_end
 	mov bp, sp
 
 	lea si, REAL_MODE_MSG
 	call print_string_16
+
+	# Load the kernel from boot drive into es:bx. In our case it is 0:0x1000
+	lea si, READ_KERNEL_MSG
+	call print_string_16
+
+	mov dl, [BOOT_DRIVE]
+	xor ax, ax
+	mov es, ax
```

Comment:
Why not xor with itself? Less state

Reply:
can't do operations on segment register with another segment register, no opcode specified for it